src/TLuaInterpreter.cpp

```diff
@@ -17659,3 +17664,116 @@ int TLuaInterpreter::showNotification(lua_State* L)
+{
+    auto& host = getHostFromLua(L);
+    if (!lua_isstring(L, 1)) {
+        lua_pushfstring(L, "registerMapInfo: bad argument #1 type (label as string expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
+    }
+
+    if (!lua_isfunction(L, 2)) {
+        lua_pushfstring(L, "registerMapInfo: bad argument #2 type (callback as function expected, got %s!)", luaL_typename(L, 2));
+        return lua_error(L);
+    }
+
+    auto name = lua_tostring(L, 1);
+    int callback = luaL_ref(L, LUA_REGISTRYINDEX);
+
+    host.mpMap->mMapInfoContributorManager->registerContributor(name, [=](int roomID, int selectionSize, int areaId, int displayAreaId, QColor& infoColor) {
+        lua_rawgeti(L, LUA_REGISTRYINDEX, callback);
+        if (roomID > 0) {
+            lua_pushinteger(L, roomID);
+        } else {
+            lua_pushnil(L);
+        }
+        lua_pushinteger(L, selectionSize);
+        lua_pushinteger(L, areaId);
+        lua_pushinteger(L, displayAreaId);
+        
+        lua_pcall(L, 4, 6, 0);
+
+        auto nResult = lua_gettop(L);
+        auto index = -nResult;
+        QString text = lua_tostring(L, index);
+        bool isBold = lua_toboolean(L, ++index);
+        bool isItalic = lua_toboolean(L, ++index);
+        int r = -1;
+        int g = -1;
+        int b = -1;
+        if (!lua_isnil(L, ++index)) {
+            r = lua_tonumber(L, index);
+        }
+        if (!lua_isnil(L, ++index)) {
+            g = lua_tonumber(L, index);
+        }
+        if (!lua_isnil(L, ++index)) {
+            b = lua_tonumber(L, index);
+        }
+        QColor color;
+        if (r >= 0 && r <= 255 && g >= 0 && g <= 255 && b >= 0 && b <= 255) {
+            color = QColor(r, g, b);
+        }
+        lua_pop(L, nResult);
+        return MapInfoProperties{ isBold, isItalic, text, color };
+    });
+    host.mpMap->mpMapper->updateInfoContributors();
+
+    lua_pushboolean(L, true);
+    return 1;
+}
+
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#killMapInfo
+int TLuaInterpreter::killMapInfo(lua_State* L)
+{
+    auto& host = getHostFromLua(L);
+    if (!lua_isstring(L, 1)) {
+        lua_pushfstring(L, "killMapInfo: bad argument #1 type (label as string expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
+    }
+
+    auto name = lua_tostring(L, 1);
+    
+    host.mpMap->mMapInfoContributorManager->removeContributor(name);
+    host.mpMap->mpMapper->updateInfoContributors();
+    lua_pushboolean(L, true);
+    return 1;
+}
+
+// Documentation: https://wiki.mudlet.org/w/Manual:Lua_Functions#enableMapInfo
+int TLuaInterpreter::enableMapInfo(lua_State* L)
+{
+    if (!lua_isstring(L, 1)) {
+        lua_pushfstring(L, "enableMapInfo: bad argument #1 type (label as string expected, got %s!)", luaL_typename(L, 1));
+        return lua_error(L);
+    }
+
+    auto name = lua_tostring(L, 1);
+    auto& host = getHostFromLua(L);
+    host.mMapInfoContributors.insert(name);
```

Comment:
Same here and for `disable` below, enabling one that doesn't exist should return nil+msg, which'll show up in debug thanks to https://github.com/Mudlet/Mudlet/pull/4678. Makes debugging easier if this information isn't silent, right